### PR TITLE
Update the contribution instructions

### PR DESCRIPTION
**Pull request by rolfedh**
Fixes https://github.com/redhat-documentation/modular-docs/issues/225
Supersedes: https://github.com/redhat-documentation/modular-docs/pull/226

### CODE REVIEW

**Comment by rolfedh** [comment]
@mjahoda @Jenny-Anne. I had to start a new branch because of some configuration issues with my fork, which was using the obsolete `master` branch. Would you please review this new PR?
I'm proposing a completely revamped set of instructions. WDYT?
Preview link: https://github.com/redhat-documentation/modular-docs/blob/cddc4c09642be123debac3907083a76ad6ad56f7/CONTRIBUTING.md

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,61 +1,101 @@
...
-STEP 5 - Create a new pull request from the GitHub web interface.
+1. **Edit content** or add new material as needed.
 
-STEP 6 - Everyone on the project team will review the merge request and add comments in GitHub. This review process will be open for one week from the day the merge request was submitted. If the merge request is still being actively discussed beyond the one week timeframe, then the merge request will stay open. Once the merge request discussion is resolved, the merge request will be NACK'd or ACK'd based on the comments given.  If no comments are given after a week, then the merge request will be ACK'd.
+2. **Stage, commit, and push** your updates:
+
+   ```bash
+   git add <path_to_file>
+   git commit -m "<your_commit_message_here>"
+   git push -f origin HEAD
```
Comment (mjahoda) [inline CONTRIBUTING.md:77]: I'm not sure about this. I would consider to simplify the first two commands by merging them into one:
`git commit -am "<your_commit_message_here>"`
Then, I believe that in our case, a simple `git push` is enough (when you locally configured Git to automatically set up remote), or:
` git push --set-upstream origin <new_branch_name>`.
Or do we need to force push?

**Review by mjahoda — CHANGES_REQUESTED** [review]
I may be wrong, but please take a look at my suggestions.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,61 +1,101 @@
...
-    git add modular-doc-manual/master.adoc
-    git commit -m -s "Updated the master.adoc file"
+* **Prepare your main branch**:
 
-STEP 4 - Push changes to the remote GitHub repository:
+   ```bash
+   git checkout main
+   git fetch upstream main
+   git rebase upstream/main  
+   git checkout -b <new_branch_name>
```
[the remark this answers is not in the document]
Comment (rolfedh) [reply CONTRIBUTING.md:59]: Nice! Thanks for sharing. Updated.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,61 +1,101 @@
...
+   ```
 
-    git push origin <new_branch_name>
+   Example:
 
-Example:
+   ```bash
+   git checkout -b new_edits
```
[the remark this answers is not in the document]
Comment (rolfedh) [reply CONTRIBUTING.md:65]: Updated. Thanks.

File: CONTRIBUTING.md (hunk 1 above)
Comment (rolfedh) [reply CONTRIBUTING.md:77]: I believe `git commit -am "<your_commit_message_here>"` only operates on already-tracked files. For untracked files, the user must still use `git add <filename>` before commiting the changes.

Comment (rolfedh) [reply CONTRIBUTING.md:77]: > when you locally configured Git to automatically set up remote

I'm not sure we can rely on this being true in most cases.

Comment (rolfedh) [reply CONTRIBUTING.md:77]: > `git push --set-upstream origin <new_branch_name>`

also works but is longer than `git push origin HEAD`, which I beleive most writers are familiar with.

Comment (rolfedh) [reply CONTRIBUTING.md:77]: > Or do we need to force push?

We don't need to force push the first time, but it's harmless. I've removed it.

**Comment by rolfedh** [comment]
I believe GitHub may have merged and closed this pull request automatically as a result of some of my command operations. Not intentional on my part.

**Comment by Jenny-Anne** [comment]
For what it's worth, the preview looks good to me :) I didn't do the step to configure an upstream remote when I set up my local repository. Is that a necessary step? It might be worth running through the process in the mod docs call. 

Also, I didn't realise that uncommented requests were automatically approved after one week as per the last line - good to know.

Thanks again for updating the page.

**Comment by rolfedh** [comment]
> configure an upstream remote

Yes. This is a required step for creating pull requests when you want to contribute to the repo. I'll be glad to demo this any time.

> uncommented requests were automatically approved after one week as per the last line 

Yes. This was new to me, too. I'm not sure we're following this in practice. I preserved that information from the original version because it seemed like a team decision. We should probably discuss whether this is still true. I've added it to the agenda for discussion.

**Pull request merged**